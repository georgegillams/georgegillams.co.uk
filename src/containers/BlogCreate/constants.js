export const KEY = 'blog-list';
export const CREATE_BLOG = `app/${KEY}/CREATE_BLOG`;
