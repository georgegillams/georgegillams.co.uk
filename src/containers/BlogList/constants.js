export const KEY = 'blog-list';
export const LOAD_BLOGS = `app/${KEY}/LOAD_BLOGS`;
export const DELETE_BLOG = `app/${KEY}/DELETE_BLOG`;
