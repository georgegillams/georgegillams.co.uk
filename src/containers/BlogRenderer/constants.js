export const KEY = 'blog-renderer';
export const LOAD_BLOG = `app/${KEY}/LOAD_BLOG`;
export const UPDATE_BLOG = `app/${KEY}/UPDATE_BLOG`;
