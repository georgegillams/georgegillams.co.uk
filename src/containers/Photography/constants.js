export const KEY = 'photography';
export const LOAD_PHOTOS = `app/${KEY}/LOAD_PHOTOS`;
