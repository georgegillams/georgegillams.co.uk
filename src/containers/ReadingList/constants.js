export const KEY = 'reading-list';
export const LOAD_BOOKS = `app/${KEY}/LOAD_BOOKS`;
export const UPDATE_BOOK = `app/${KEY}/UPDATE_BOOK`;
export const DELETE_BOOK = `app/${KEY}/DELETE_BOOK`;
