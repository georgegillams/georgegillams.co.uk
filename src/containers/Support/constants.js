export const KEY = 'support';
export const LOAD_LINKS = `app/${KEY}/LOAD_LINKS`;
export const CREATE_LINK = `app/${KEY}/CREATE_LINK`;
export const DELETE_LINK = `app/${KEY}/DELETE_LINK`;
