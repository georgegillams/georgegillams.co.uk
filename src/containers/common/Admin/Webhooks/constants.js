export const KEY = 'webhooks';

export const LOAD_ENDPOINTS = `app/${KEY}/LOAD_ENDPOINTS`;
export const CREATE_ENDPOINT = `app/${KEY}/CREATE_ENDPOINT`;
export const UPDATE_ENDPOINT = `app/${KEY}/UPDATE_ENDPOINT`;
export const REMOVE_ENDPOINT = `app/${KEY}/REMOVE_ENDPOINT `;

export const LOAD_NOTIFICATIONS = `app/${KEY}/LOAD_NOTIFICATIONS`;
export const REMOVE_NOTIFICATION = `app/${KEY}/REMOVE_NOTIFICATION`;
