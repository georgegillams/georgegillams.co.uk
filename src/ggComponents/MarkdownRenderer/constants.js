const DEFAULT_SUPPORTED_FEATURES = [
  'code',
  'blockCode',
  'bold',
  'italic',
  'strikethrough',
  'quotation',
  'bigLink',
  'link',
  'footnote',
  'image',
  'video',
  'citation',
  'sectioning',
  'linebreak',
  'bulletList',
  'numberedList',
];

export default DEFAULT_SUPPORTED_FEATURES;
export { DEFAULT_SUPPORTED_FEATURES };
