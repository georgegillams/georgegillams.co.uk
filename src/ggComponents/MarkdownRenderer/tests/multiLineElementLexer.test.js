#!/usr/bin/env node

import {
  parseMultiLineElements,
  parseLinesForQuoteBlock,
  parseLinesForBulletList,
  parseLinesForBlockCode,
} from '../multiLineElementLexer.js';
import { DEFAULT_SUPPORTED_FEATURES } from '../constants.js';

// #region individual lists
test('parses lines to insert code-block', () => {
  const result = parseLinesForBlockCode([
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: '```' },
    { type: 'line', content: ' - Some code' },
    {
      type: 'line',
      content: ' - Some code with a [link](https://duckduckgo.com)',
    },
    { type: 'line', content: '```' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: 'Line 3' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('parses lines to insert code-block with language', () => {
  const result = parseLinesForBlockCode([
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: '``` js' },
    { type: 'line', content: ' - Some code' },
    {
      type: 'line',
      content: ' - Some code with a [link](https://duckduckgo.com)',
    },
    { type: 'line', content: '```' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: 'Line 3' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('parses lines to insert code-block', () => {
  const result = parseLinesForBlockCode([
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: '``` js, https://github.com/' },
    { type: 'line', content: ' - Some code' },
    {
      type: 'line',
      content: ' - Some code with a [link](https://duckduckgo.com)',
    },
    { type: 'line', content: '```' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: 'Line 3' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('parses lines to insert bullet-list', () => {
  const result = parseLinesForBulletList([
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: ' - Bullet 1' },
    { type: 'line', content: ' - Bullet 2' },
    { type: 'line', content: ' - Bullet 3' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: 'Line 3' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('parses lines to insert numbered-list', () => {
  const result = parseLinesForQuoteBlock([
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: ' 1. Numbered item 1' },
    { type: 'line', content: ' 2. Numbered item 2' },
    { type: 'line', content: ' 3. Numbered item 3' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: 'Line 3' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('parses lines containing no comment block', () => {
  const result = parseLinesForQuoteBlock([
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: 'Line 3' },
    { type: 'line', content: 'Line 4' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('parses lines to insert comment block at start', () => {
  const result = parseLinesForQuoteBlock([
    { type: 'line', content: '> Quote line 1' },
    { type: 'line', content: '> Quote line 2' },
    { type: 'line', content: '> Quote line 3' },
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: 'Line 3' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('parses lines to insert comment block in middle', () => {
  const result = parseLinesForQuoteBlock([
    { type: 'line', content: 'Line 1' },
    { type: 'line', content: 'Line 2' },
    { type: 'line', content: '> Quote line 1' },
    { type: 'line', content: '> Quote line 2' },
    { type: 'line', content: '> Quote line 3' },
    { type: 'line', content: 'Line 3' },
  ]);
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
// #endregion individual lists

// #region structure
test('parses comment block inside section', () => {
  const result = parseMultiLineElements(
    [
      { type: 'line', content: 'Line 1' },
      {
        type: 'section',
        children: [
          { type: 'line', content: 'Line 2' },
          { type: 'line', content: '> Quote line 1' },
          { type: 'line', content: '> Quote line 2' },
          { type: 'line', content: '> Quote line 3' },
          { type: 'line', content: 'Line 3' },
        ],
      },
      { type: 'line', content: 'Line 4' },
    ],
    DEFAULT_SUPPORTED_FEATURES
  );
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});

test('ignores comment block if feature is unsupported', () => {
  const result = parseMultiLineElements(
    [
      { type: 'line', content: 'Line 1' },
      { type: 'line', content: 'Line 2' },
      { type: 'line', content: '> Unsupported quote line 1' },
      { type: 'line', content: '> Unsupported quote line 2' },
      { type: 'line', content: '> Unsupported quote line 3' },
      { type: 'line', content: 'Line 3' },
    ],
    []
  );
  expect(result.error).toBe(undefined);
  expect(result).toMatchSnapshot();
});
// #endregion structure
