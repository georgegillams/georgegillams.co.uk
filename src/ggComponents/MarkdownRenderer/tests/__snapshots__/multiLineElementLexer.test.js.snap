// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ignores comment block if feature is unsupported 1`] = `
Array [
  Object {
    "children": Array [
      "Line 1",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      "Line 2",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      "> Unsupported quote line 1",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      "> Unsupported quote line 2",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      "> Unsupported quote line 3",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      "Line 3",
    ],
    "type": "paragraph",
  },
]
`;

exports[`parses comment block inside section 1`] = `
Array [
  Object {
    "children": Array [
      "Line 1",
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          "Line 2",
        ],
        "type": "paragraph",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              "Quote line 1",
            ],
            "type": "paragraph",
          },
          Object {
            "children": Array [
              "Quote line 2",
            ],
            "type": "paragraph",
          },
          Object {
            "children": Array [
              "Quote line 3",
            ],
            "type": "paragraph",
          },
        ],
        "type": "quotation",
      },
      Object {
        "children": Array [
          "Line 3",
        ],
        "type": "paragraph",
      },
    ],
    "type": "section",
  },
  Object {
    "children": Array [
      "Line 4",
    ],
    "type": "paragraph",
  },
]
`;

exports[`parses lines containing no comment block 1`] = `
Array [
  Object {
    "content": "Line 1",
    "type": "line",
  },
  Object {
    "content": "Line 2",
    "type": "line",
  },
  Object {
    "content": "Line 3",
    "type": "line",
  },
  Object {
    "content": "Line 4",
    "type": "line",
  },
]
`;

exports[`parses lines to insert bullet-list 1`] = `
Array [
  Object {
    "content": "Line 1",
    "type": "line",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          "Bullet 1",
        ],
        "type": "paragraph",
      },
      Object {
        "children": Array [
          "Bullet 2",
        ],
        "type": "paragraph",
      },
      Object {
        "children": Array [
          "Bullet 3",
        ],
        "type": "paragraph",
      },
    ],
    "type": "bulletList",
  },
  Object {
    "content": "Line 2",
    "type": "line",
  },
  Object {
    "content": "Line 3",
    "type": "line",
  },
]
`;

exports[`parses lines to insert code-block 1`] = `
Array [
  Object {
    "content": "Line 1",
    "type": "line",
  },
  Object {
    "codeLines": Array [
      " - Some code",
      " - Some code with a [link](https://duckduckgo.com)",
    ],
    "language": "",
    "type": "blockCode",
  },
  Object {
    "content": "Line 2",
    "type": "line",
  },
  Object {
    "content": "Line 3",
    "type": "line",
  },
]
`;

exports[`parses lines to insert code-block 2`] = `
Array [
  Object {
    "content": "Line 1",
    "type": "line",
  },
  Object {
    "codeLines": Array [
      " - Some code",
      " - Some code with a [link](https://duckduckgo.com)",
    ],
    "language": "js",
    "type": "blockCode",
    "url": "https://github.com/",
  },
  Object {
    "content": "Line 2",
    "type": "line",
  },
  Object {
    "content": "Line 3",
    "type": "line",
  },
]
`;

exports[`parses lines to insert code-block with language 1`] = `
Array [
  Object {
    "content": "Line 1",
    "type": "line",
  },
  Object {
    "codeLines": Array [
      " - Some code",
      " - Some code with a [link](https://duckduckgo.com)",
    ],
    "language": "js",
    "type": "blockCode",
  },
  Object {
    "content": "Line 2",
    "type": "line",
  },
  Object {
    "content": "Line 3",
    "type": "line",
  },
]
`;

exports[`parses lines to insert comment block at start 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          "Quote line 1",
        ],
        "type": "paragraph",
      },
      Object {
        "children": Array [
          "Quote line 2",
        ],
        "type": "paragraph",
      },
      Object {
        "children": Array [
          "Quote line 3",
        ],
        "type": "paragraph",
      },
    ],
    "type": "quotation",
  },
  Object {
    "content": "Line 1",
    "type": "line",
  },
  Object {
    "content": "Line 2",
    "type": "line",
  },
  Object {
    "content": "Line 3",
    "type": "line",
  },
]
`;

exports[`parses lines to insert comment block in middle 1`] = `
Array [
  Object {
    "content": "Line 1",
    "type": "line",
  },
  Object {
    "content": "Line 2",
    "type": "line",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          "Quote line 1",
        ],
        "type": "paragraph",
      },
      Object {
        "children": Array [
          "Quote line 2",
        ],
        "type": "paragraph",
      },
      Object {
        "children": Array [
          "Quote line 3",
        ],
        "type": "paragraph",
      },
    ],
    "type": "quotation",
  },
  Object {
    "content": "Line 3",
    "type": "line",
  },
]
`;

exports[`parses lines to insert numbered-list 1`] = `
Array [
  Object {
    "content": "Line 1",
    "type": "line",
  },
  Object {
    "content": " 1. Numbered item 1",
    "type": "line",
  },
  Object {
    "content": " 2. Numbered item 2",
    "type": "line",
  },
  Object {
    "content": " 3. Numbered item 3",
    "type": "line",
  },
  Object {
    "content": "Line 2",
    "type": "line",
  },
  Object {
    "content": "Line 3",
    "type": "line",
  },
]
`;
